### PR TITLE
Use the `soft7` package

### LINKED ISSUE #34: ✨ Use SOFT7
Use [SOFT7](https://github.com/SINTEF/soft7).

### PR DESCRIPTION
**Pull request by CasperWA**
Closes #34 

Implement usage of the [`soft7`](https://github.com/SINTEF/soft7) package.

As a "breaking" change, this means the `uri` (or now `identity`) value is no longer so strictly guarded and validated. There is now no error if the namespace or composition is malformed according to the TEAM4.0-style URLs.

~The `soft7` package is temporarily installed directly from GitHub, specifically the `cwa/close-7-unify-with-dlite` branch, which is the head branch for PR SINTEF/soft7#51.~ v0.2.0 (at minimum) is now used for the `soft7` package.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ COPY pyproject.toml LICENSE README.md ./
 
 # Install dependencies
 RUN apt-get update && \
-  apt-get install -y procps && \
+  apt-get install -y git procps && \
```
Comment (quaat) [inline Dockerfile:10]: Why do we need git in the docker container? Should we add a new development layer?

File: tests/conftest.py [hunk 2]
```diff
@@ -283,13 +283,12 @@ def pytest_generate_tests(metafunc: pytest.Metafunc) -> None:
...
 
     import re
-    from copy import deepcopy
     from pathlib import Path
 
     import yaml
 
     def get_version_name(uri: str) -> tuple[str, str]:
-        """Return the version and name part of a uri."""
+        """Return the version and name part of a TEAM4.0-style URL."""
```
Comment (quaat) [inline tests/conftest.py:291]: "TEAM4.0-style URL" -- is that a thing? :question:

File: tests/service/backend/test_mongodb.py [hunk 3]
```diff
@@ -93,15 +93,15 @@ def test_create(mongo_backend: GetMongoBackend, parameterized_entity: Parameteri
...
 
     assert isinstance(entity_from_backend, dict)
-    assert entity_from_backend == parameterized_entity.backend_entity
+    assert entity_from_backend == parameterized_entity.parsed_entity
 
     # Create multiple entities
     raw_entities = [
         {
-            "uri": "http://onto-ns.com/meta/1.0/Test",
+            "identity": "http://onto-ns.com/meta/1.0/Test",
```
Comment (quaat) [inline tests/service/backend/test_mongodb.py:104]: Try adding an identity which is not a URI

**Review by quaat — APPROVED** [review]
Minor comments added.

File: Dockerfile (hunk 1 above)
Comment (CasperWA) [reply Dockerfile:10]: As long as we are not running pre-commit, we do not need `git`, no. I'll remove it.

File: tests/conftest.py (hunk 2 above)
Comment (CasperWA) [reply tests/conftest.py:291]: I meant the <namespace>/<version>/<name> and URI's always being resolvable URLs as well, usually (but not always) starting with `http://onto-ns.com/meta/`.

File: tests/service/backend/test_mongodb.py (hunk 3 above)
Comment (CasperWA) [reply tests/service/backend/test_mongodb.py:104]: As a test, you mean? Sure.

**Pull request merged**